moveit_commander/src/moveit_commander/move_group.py

```diff
@@ -597,14 +598,21 @@ def go(self, joints=None, wait=True):
             except TypeError:
                 self.set_joint_value_target(joints)
         if wait:
-            return self._g.move()
+            if planning_pipeline and planner_id:
+                return self._g.move(planning_pipeline, planner_id)
+            else:
+                return self._g.move()
```

Comment:
`mgc.go(joints= ..., planning_pipeline='OMPL')` should try to use the default planner of the pipeline (OMPL) and not silently revert to use the default pipeline.

Reply:
There is no default per pipeline, so I went with a more verbose error message instead.

Reply:
That's not true. [The OMPL plugin even supports defaults per group](https://github.com/ros-planning/moveit/blob/master/moveit_planners/ompl/ompl_interface/src/planning_context_manager.cpp#L436-L456) and sending an empty `planner_id` triggers the default. A similar functionality is not implemented for `pilz_industrial_motion_planner` at the moment, but could surely be added there.

I believe rejecting an empty `planner_id` should be the responsibility of the respective planner manager, not of the front end.
The other way around (an empty pipeline, but non-empty planner_id) also makes sense for me as a way to set the planner for the currently selected pipeline.
I would expect both cases to be supported instead of throwing errors. :question:

Reply:
Admittedly I didn't look into it too deeply, I just looked at the ROS parameters to check for a default.

But since there is a `planner_id_` setting in the `move_group_interface` which is filled in by default, the planning plugin wouldn't receive a request with an empty `planner_id` field anyway. If we want to send the MotionPlanRequest with an empty `planner_id` field, the `plan` and `move` functions need to have a different default value than an empty string. Is that better? See the newest commit for how that would look.

I would just impose the (imo minor) restriction "set both id and pipeline to use this interface", and wrap it in `move_lin`, `move_ptp`, `move_ompl` functions myself.

Reply:
> If we want to send the MotionPlanRequest with an empty planner_id field, the plan and move functions need to have a different default value than an empty string. Is that better?

No you are right. In C++, this is terminally ugly. My initial comment only referred to the python API though.
For C++, you would usually have different method names `planWithPipeline(string pipeline, string planner="")` and `planWithPlanner(string planner)` instead.
If you think this is a lousy implementation as well, you can still enforce your set-pipeline-and-planner constraint in the C++ API... Either way we should get this merged.